vignettes/adtr.Rmd

```diff
@@ -0,0 +1,725 @@
+
+In the below example, we use `derive_var_merged_summary()` function multiple
+times in order to process additional variables for sum of diameter parameter,
+and calculate the analysis flag (`ANL01FL`).
+
+In the first step we concatenate lesions IDS measured `LSEXP` at baseline, and merge in with the second step where we concatenate the lesions IDS assessed `LSASS` at that visit record.  You can then compare `LSEXP` vs `LSASS` at each visit.
+
+In the final step, the analysis flag is set `ANL01FL = "Y"` where lesions
+assessed at post-baseline match baseline.
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  derive_var_merged_summary(
+    dataset_add = adtr,
+    by_vars = exprs(USUBJID),
+    filter_add = AVISIT == "BASELINE" &
+      ((str_starts(PARAMCD, "LDIAM") & TULOCGR1 == "NON-NODAL") |
+        (str_starts(PARAMCD, "NLDIAM") & TULOCGR1 == "NODAL")),
+    new_var = LSEXP,
+    analysis_var = TRLNKID,
+    summary_fun = function(x) paste(sort(x), collapse = ", ")
+  ) %>%
+  derive_var_merged_summary(
+    dataset_add = adtr,
+    by_vars = exprs(USUBJID, AVISIT),
+    filter_add = ((str_starts(PARAMCD, "LDIAM") & TULOCGR1 == "NON-NODAL") |
+      (str_starts(PARAMCD, "NLDIAM") & TULOCGR1 == "NODAL")) & ANL01FL == "Y",
+    new_var = LSASS,
+    analysis_var = TRLNKID,
+    summary_fun = function(x) paste(sort(x), collapse = ", ")
+  ) %>%
+  mutate(
+    ANL01FL = if_else(LSEXP == LSASS, "Y", NA_character_)
+  )
+```
+
+In the next step we include the analysis day and date (`ADY`/`ADT`) variables to
+the sum of diameter parameter dataset.  If there are different analysis
+day/dates associated with a particular visit, we take the minimum day/date.
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  derive_var_merged_summary(
+    dataset_add = adtr,
+    by_vars = exprs(USUBJID, AVISIT),
+    filter_add = (str_starts(PARAMCD, "LDIAM") & TULOCGR1 == "NON-NODAL") |
+      (str_starts(PARAMCD, "NLDIAM") & TULOCGR1 == "NODAL"),
+    new_var = ADY,
+    analysis_var = ADY,
+    summary_fun = function(x) min(x, na.rm = TRUE)
+  ) %>%
+  derive_var_merged_summary(
+    dataset_add = adtr,
+    by_vars = exprs(USUBJID, AVISIT),
+    filter_add = (str_starts(PARAMCD, "LDIAM") & TULOCGR1 == "NON-NODAL") |
+      (str_starts(PARAMCD, "NLDIAM") & TULOCGR1 == "NODAL"),
+    new_var = ADT,
+    analysis_var = ADT,
+    summary_fun = function(x) min(x, na.rm = TRUE)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = exprs(USUBJID, PARAMCD, AVISIT, LSEXP, LSASS, ANL01FL, ADT, ADY)
+)
+```  
+
+
+## Derive Baseline (`ABLFL`, `BASE`) {#base}
+
+These functions are available in `admiral`, below examples show its usage for `ADTR`. Here we create baseline record flag `ABLFL` by selecting all records prior to analysis day 1, and then derive the baseline value `BASE`.
```

Comment:
As for baseline we are assuming all lesions at ady<=1 are assessed are at baseline visit. But I have seen some time lesions are also recorded at screening. So, I suggest to add a statement in the end " if in a particular study/or subject, lesions are also collected at multiple visits before day 1, so make sure to update appropriate filter condition."

Reply:
The text was not 100% precise. The function call does not select _all_ records but the _last_ record prior to analysis day 1. I have updated the paragraph.

Please have a look.